service/internal/proctelemetry/config.go

```diff
@@ -71,7 +71,9 @@ func InitMetricReader(ctx context.Context, reader config.MetricReader, asyncErro
 		return initPullExporter(reader.Pull.Exporter, asyncErrorChannel)
 	}
 	if reader.Periodic != nil {
-		opts := []sdkmetric.PeriodicReaderOption{}
+		opts := []sdkmetric.PeriodicReaderOption{
+			sdkmetric.WithProducer(opencensus.NewMetricProducer()),
```

Comment:
I dont think I understand what `WithProducer` does, how was this doing anything before? Or has it been very broken without this?

Reply:
it has been very broken for periodic metric readers that cares about components that produce opencensus metrics.